packages/storage/src/indexed-db/updater.ts

```diff
@@ -41,15 +41,12 @@ export class IbdUpdater {
     const tables = updates.all.map(u => u.table);
     const tx = this.db.transaction(tables, 'readwrite');
 
-    const batchOperations = [];
-
     // Batch all the updates into promises
     for (const update of updates.all) {
-      const putOperation = tx.objectStore(update.table).put(update.value, update.key);
-      batchOperations.push(putOperation);
+      tx.objectStore(update.table).put(update.value, update.key);
```

Comment:
Gabe rightly identified that performing an await inside IndexedDB transaction can risk closing the transaction prematurely in the event loop, which we shouldn't do.